core/src/main/java/org/elasticsearch/action/delete/DeleteRequest.java

```diff
@@ -221,4 +222,34 @@ public void writeTo(StreamOutput out) throws IOException {
     public String toString() {
         return "delete {[" + index + "][" + type + "][" + id + "]}";
     }
+
+    /**
+     * Override this method from ReplicationAction, this is where we are storing our state in the request object (which we really shouldn't
+     * do). Once the transport client goes away we can move away from making this available, but in the meantime this is dangerous to set or
+     * use because the DeleteRequest object will always be wrapped in a bulk request envelope, which is where this *should* be set.
+     */
+    @Override
+    public long primaryTerm() {
+        throw new UnsupportedOperationException("primary term should never be set on DeleteRequest");
+    }
+
+    /**
+     * Override this method from ReplicationAction, this is where we are storing our state in the request object (which we really shouldn't
+     * do). Once the transport client goes away we can move away from making this available, but in the meantime this is dangerous to set or
+     * use because the DeleteRequest object will always be wrapped in a bulk request envelope, which is where this *should* be set.
+     */
+    @Override
+    public void primaryTerm(long term) {
+        throw new UnsupportedOperationException("primary should never be set on DeleteRequest");
```

Comment:
Sorry, I wasn't explicit, but you're missing "term" everywhere except for the ones that you just fixed.